### PR TITLE
Remove check for DRep metadata size

### PR DESCRIPTION
**Pull request by CarlosLopezDeLara**
# Changelog

```yaml
- description: |
    https://cips.cardano.org/cip/CIP-0119 proposes using a schema where the basic jsonld structure is heavier than 512. The agreement is that DBsync will set a limit for the size of metadata that is downloaded into the DB, but API or CLI do not need to impose any restrictions.
# uncomment types applicable to the change:
  type:
   - feature        # introduces a new feature
   - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - refactoring    # QoL changes
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

https://cips.cardano.org/cip/CIP-0119 proposes using a schema where the basic jsonld structure is heavier than 512.
https://github.com/cardano-foundation/CIPs/pull/835

# How to trust this PR

Highlight important bits of the PR that will make the review faster. If there are commands the reviewer can run to observe the new behavior, describe them.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

**Review by carbolymer — APPROVED** [review]

File: cardano-api/internal/Cardano/Api/DRepMetadata.hs [hunk 1]
```diff
@@ -57,37 +56,13 @@ instance SerialiseAsRawBytes (Hash DRepMetadata) where
...
-  -- TODO confirm if there are size limits to the DRep metadata
-  | BS.length bs <= 512 = do
-      let md = DRepMetadata bs
-      let mdh = DRepMetadataHash (Crypto.hashWith id bs)
-      return (md, mdh)
-  | otherwise = Left $ DRepMetadataInvalidLengthError 512 (BS.length bs)
+  -> (DRepMetadata, Hash DRepMetadata)
+hashDRepMetadata bs =
+  let md  = DRepMetadata bs
+      mdh = DRepMetadataHash (Crypto.hashWith id bs) in
```
[the remark this answers is not in the document]
Comment (CarlosLopezDeLara) [reply cardano-api/internal/Cardano/Api/DRepMetadata.hs:63]: I think we just want to get the hash of whatever the uses passes. 

Typically the user would: 

1. upload the metadata file (it might be normalized, formatted, etc at this stage depending on the hosting service), 
2. download metadata from the final URL and hash it. If we do any intermediate processing of the file it will not match what they get using an alternative method. i.e. b2sum -l 254 drep-metadata.json

[the remark this answers is not in the document]
Comment (carbolymer) [reply cardano-api/internal/Cardano/Api/DRepMetadata.hs:63]: I partially agree 😄 
When the user wants to pass any garbage, or bytes, or a cat image, or any arbitrary data - it's fine. We should hash it as it is.
However using JSON here assumes using a data structure to store information. So JSON already encodes data and ignores white space. So the user might want to replace tabs with spaces in their metadata JSON - and they will get a different hash!
It may be a problem. It is also a bit different question - is it our problem?
Does it also affect backwards compatibility?

[the remark this answers is not in the document]
Comment (smelc) [reply cardano-api/internal/Cardano/Api/DRepMetadata.hs:63]: I see your point @carbolymer but I would refrain from normalizing the JSON because we would be at risk of becoming inconsistent with other tools like `b2sum`, that don't treat JSON in any special manner.

IMHO this change doesn't break backwards compatibility since we allow more inputs now. If we allowed _less_ inputs, it would be a breaking change.

**Review by smelc — APPROVED** [review]

File: cardano-api/internal/Cardano/Api/DRepMetadata.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (CarlosLopezDeLara) [reply cardano-api/internal/Cardano/Api/DRepMetadata.hs:63]: A similar conversation that went on for weeks around various CIPs related to Governance metadata ->
https://github.com/cardano-foundation/CIPs/pull/835

TLDR -> 

@disasm:  

> I'm in agreement with @kderme and @gitmachtl . We don't want to canonicalize json because it can cause collisions. This was a major point @dcoutts made when we stopped using canonical json after Byron. It's a security risk to massage data before hashing.

If we fiddle with the data, then it's our problem if we get it wrong. The data correctness and format is responsibility of the user, we just need to do the hashing right. Our output matches the output of `b2sum -l 256 <file>` which is what we are meant to do.

[the remark this answers is not in the document]
Comment (carbolymer) [reply cardano-api/internal/Cardano/Api/DRepMetadata.hs:63]: Thanks for clarifying things out. I'm ok with how it is implemented right now.

>We don't want to canonicalize json because it can cause collisions.

I don't see how canonicalization can cause more collisions though.
